zebra-test/src/network_addr.rs

```diff
@@ -0,0 +1,303 @@
+use lazy_static::lazy_static;
+
+lazy_static! {
+    /// Array of `addr` (v1) test vectors containing IP addresses.
+    ///
+    /// These test vectors can be read by [`zebra_network::protocol::external::Codec::read_addr`].
+    /// They should produce successful results containing IP addresses.
+    // From https://github.com/zingolabs/zcash/blob/9ee66e423a3fbf4829ffeec354e82f4fbceff864/src/test/netbase_tests.cpp#L397
+    pub static ref ADDR_V1_IP_VECTORS: Vec<Vec<u8>> = vec![
+        // stream_addrv1_hex
+        <Vec<u8>>::from_hex(
+            concat!(
+                "02", // number of entries
+
+                "61bc6649",                         // time, Fri Jan  9 02:54:25 UTC 2009
+                "0000000000000000",                 // service flags, NODE_NONE
+                "00000000000000000000000000000001", // address, fixed 16 bytes (IPv4 embedded in IPv6)
+                "0000",                             // port
+
+                "79627683",                         // time, Tue Nov 22 11:22:33 UTC 2039
+                "0100000000000000",                 // service flags, NODE_NETWORK
+                "00000000000000000000000000000001", // address, fixed 16 bytes (IPv6)
+                "00f1",                             // port
+            )
+        ).expect("Message bytes are in valid hex representation"),
+
+        // stream_torv3_incompatibly_serialized_to_v1
+        <Vec<u8>>::from_hex(
+            concat!(
+                "01", // number of entries
+
+                "79627683",                         // time, Tue Nov 22 11:22:33 UTC 2039
+                "0100000000000000",                 // service flags, NODE_NETWORK
+                "00000000000000000000000000000000", // address, fixed 16 bytes (IPv6)
+                "235a",                             // port
+            )
+        ).expect("Message bytes are in valid hex representation"),
+
+        // Extra test cases:
+        //
+        // all services flags set
+        <Vec<u8>>::from_hex(
+            concat!(
+                "01", // number of entries
+
+                "61bc6649",                         // time, Fri Jan  9 02:54:25 UTC 2009
+                "ffffffffffffffff",                 // service flags, all set
+                "00000000000000000000000000000001", // address, fixed 16 bytes (IPv4 embedded in IPv6)
+                "0000",                             // port
+            )
+        ).expect("Message bytes are in valid hex representation"),
+
+        // IPv4
+        <Vec<u8>>::from_hex(
+            concat!(
+                "01", // number of entries
+
+                "61bc6649",                         // time, Fri Jan  9 02:54:25 UTC 2009
+                "0100000000000000",                 // service flags, NODE_NETWORK
+                "00000000000000000000ffff",         // address, fixed 16 bytes (IPv4 embedded in IPv6)
+                "7f000001",                         // IPv4, 127.0.0.1
```

Comment:
Nit: minor tweak for clarity:

```suggestion
                "00000000000000000000ffff",         // address, first 12 bytes of fixed 16 bytes
                                                    // (prefix for IPv4 embedded in IPv6)
                "7f000001",                         // final 4 bytes for IPv4, 127.0.0.1
```

Reply:
Fixed in a7f11c210